src/test/groovy/graphql/schema/idl/SchemaPrinterComparatorEnvironmentTest.groovy

```diff
@@ -0,0 +1,141 @@
+package graphql.schema.idl
+
+import graphql.AssertException
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLObjectType
+import spock.lang.Specification
+
+import static graphql.schema.idl.SchemaPrinterComparatorEnvironment.newEnvironment
+
+class SchemaPrinterComparatorEnvironmentTest extends Specification {
```

Comment:
Extracted out into its own test class.

A few extra tests included to cover new `transform` method and ensure `equals` and `hashCode` are implemented correctly since this type is used as the key for hashmap registry lookups.